pkgs/development/libraries/dnnl/default.nix

```
@@ -0,0 +1,56 @@
+/*
+  This library can optionally use the MKL package. This is disabled by default
+  since it has a proprietary license; to enable it, add the following config
+  variable in an overlay:
+
+  self: super: {
+    mklSupport = true;
+  }
+*/
+{ stdenv, lib, fetchFromGitHub, substituteAll, autoPatchelfHook
+, cmake, bash
+, mklSupport ? false, mkl ? null
+}:
+
+assert mklSupport -> mkl != null;
+
+stdenv.mkDerivation rec {
+  pname = "dnnl";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "mkl-dnn";
+    rev = "v${version}";
+    sha256 = "006mh3kbmr263k40vszq8ckbz6kzclqhqifwja056aw0m78rcvkz";
+  };
+
+  patches = [ (substituteAll {
+    src = ./bash-to-sh.patch;
+    inherit bash;
+  }) ];
+
+  outputs = [ "out" "dev" "doc" ];
+
+  cmakeFlags = lib.optionals mklSupport [ "-D_DNNL_USE_MKL=ON" ];
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = lib.optionals mklSupport [ mkl ];
+
+  doCheck = true;
+
+  # the test executables aren't able to find libmkldnn.so through LD_LIBRARY_PATH, so patch them.
+  checkInputs = [ autoPatchelfHook ];
+  preCheck = ''
+    addAutoPatchelfSearchPath $PWD/src
+    autoPatchelf tests examples
+  '';
+
+  meta = with lib; {
+    description = "Deep Neural Network Library (DNNL)";
+    homepage = "https://intel.github.io/mkl-dnn/dev_guide_transition_to_dnnl.html";
+    license = licenses.asl20;
+    platforms = platforms.all;
```

```suggestion
    platforms = platforms.x86;
```

adds x86 specific extensions to compiler:
```
g++: error: unrecognized command line option '-msse4.1'
g++: error: unrecognized command line option '-msse4.1'
g++: error: unrecognized command line option '-msse4.1'
```

---

Updated accordingly